cmd/vela-makisu/registry.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2020 Target Brands, Inr. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/uber/makisu/lib/registry"
+	"github.com/urfave/cli/v2"
+)
+
+const (
+	// dockerHubConf represents the config that provides authentication to docker.index.io.
+	dockerHubConf = `{
+		"index.docker.io": {
+      ".*": {
+         "security": {
+            "tls": {
+               "client": {
+                  "disabled": false
+               }
+            },
+            "basic": {
+               "username": "",
+               "password": ""
+            }
+         }
+      }
+   }		
+ }`
+
+	// mirrorConf represents the config that provides authentication to a registry mirror.
+	mirrorConf = `{
+		"%s": {
+      ".*": {
+         "security": {
+            "tls": {
+               "client": {
+                  "disabled": false
+               }
+            },
+            "basic": {
+               "username": "",
+               "password": ""
+            }
+         }
+      }
+   }		
+ }`
+
+	// registryConf represents the config that provides authentication to a registry.
+	registryConf = `{
+		"%s": {
+      ".*": {
+         "security": {
+            "tls": {
+               "client": {
+                  "disabled": false
+               }
+            },
+            "basic": {
+               "username": "%s",
+               "password": "%s"
+            }
+         }
+      }
+   }		
+ }`
+)
+
+// Registry represents the input parameters for the plugin.
+type Registry struct {
+	// full url to Docker Registry
+	Addr string
+	// enables building images without publishing to the registry
+	DryRun bool
+	// full url to a Docker Registry mirror
+	Mirror string
+	// password for communication with the Docker Registry
+	Password string
+	// user name for communication with the Docker Registry
+	Username string
+}
+
+var (
+	// appFs represents a instance of the filesystem.
+	appFS = afero.NewOsFs()
+
+	// configFlags represents for config settings on the cli.
+	// nolint
+	configFlags = []cli.Flag{
+		&cli.StringFlag{
+			EnvVars:  []string{"PARAMETER_REGISTRY", "REGISTRY_ADDR"},
+			FilePath: string("/vela/parameters/makisu/registry/name,/vela/secrets/docker/registry/name"),
+			Name:     "registry.addr",
+			Usage:    "Docker registry address to communicate with",
+			Value:    "index.docker.io",
+		},
+		&cli.BoolFlag{
+			EnvVars:  []string{"PARAMETER_DRY_RUN", "REGISTRY_DRY_RUN"},
+			FilePath: string("/vela/parameters/makisu/registry/path,/vela/secrets/makisu/registry/path,/vela/secrets/makisu/path"),
```

Comment:
Are the paths here accurate?

I wonder if they could be named a bit closer to the flag and env var naming 👍

```suggestion
			FilePath: string("/vela/parameters/makisu/registry/dry_run,/vela/secrets/makisu/registry/path,/vela/secrets/makisu/dry_run"),
```